purchase_packaging/models/purchase_order_line.py

```diff
@@ -48,38 +48,48 @@ def _get_product_seller(self):
             uom_id=self.product_uom)
 
     @api.multi
-    @api.depends('product_purchase_uom_id', 'product_purchase_qty')
+    @api.depends('product_purchase_uom_id',
+                 'product_purchase_qty',
+                 'product_purchase_uom_id.category_id')
     def _compute_product_qty(self):
         """
         Compute the total quantity
         """
+        uom_categories = self.mapped("product_purchase_uom_id.category_id")
+        uom_obj = self.env['product.uom']
+        to_uoms = uom_obj.search(
+            [('category_id',
+              'in',
+              uom_categories.ids),
+             ('uom_type', '=', 'reference')])
+        uom_by_category = {to_uom.category_id: to_uom for to_uom in to_uoms}
         for line in self:
-            uom_obj = self.env['product.uom']
-            to_uom = uom_obj.search(
-                [('category_id',
-                  '=',
-                  line.product_purchase_uom_id.category_id.id),
-                 ('uom_type', '=', 'reference')], limit=1)
             if not line.product_purchase_uom_id:
                 return
             line.product_qty = line.product_purchase_uom_id._compute_quantity(
                 line.product_purchase_qty,
-                to_uom)
+                uom_by_category.get(line.product_purchase_uom_id.category_id))
```

Comment:
disclaimer : I don't use this module and I don't know how it is supposed to work. I agree that your change is not modifying the original behavior, but I have doubts about that behavior. Why are we using the reference unit of the category of the uom as target for a conversion? In the standard stock addon, the quantities are converted to the uom of the product which makes much more sense in my opinion. Since this is the unit which will be used when processing in the move, should we not use it in here?

Reply:
cc @lmignon who is the original author of the module

Reply:
@gurneyalex Indeed, this code is correct as in this module we change the behaviour of standards Odoo fields product_qty and product_uom_id. They are computed now.

The fields that appear in forms are new ones to make conversion easier. We use here two dimensional units.

So, we compute the original fields considering reference purchase units to make procurements and stock moves values conversion transparent for user.

@Cedric-Pigeon I know you worked on this too. An advice ?